boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -100,6 +100,24 @@ CXXFLAGS := $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
 CPPFLAGS := $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS)
 AFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
+ifeq ($(CONFIG_LIBCXX),y)
+  CXXFLAGS += -std=c++14 -D_LIBCPP_STD_VER=14
+
+  ifeq ($(CONFIG_HOST_MACOS),y)
+    # macOS uses libc++abi
+    CXXFLAGS += -DLIBCXX_BUILDING_LIBCXXABI
```

Comment:
libc++ specific config should go to https://github.com/apache/incubator-nuttx/blob/master/libs/libxx/libcxx.defs?

Reply:
sure

Reply:
well, i remembered why i put this here. it's sim specific. do you prefer to have sim-specific logic in libcxx.defs?

Reply:
But, I think it's toolchain specific: libsupc++.a for gcc and libc++abi.a for clang?

Reply:
is it?
i thought it was rather os specific.
macOS uses libc++abi, other BSDs use libcxxrt, ubuntu uses libsupc++, etc.
anyway, Make.defs is the appropriate place to put toolchain-specific bits, isn't it?